src/zjs_callbacks.c

```diff
@@ -313,19 +313,9 @@ void zjs_remove_callback(int32_t id)
 
 void zjs_signal_callback(int32_t id, void* args, uint32_t size)
 {
-    if (args && size) {
-        // signal with args
-        int ret = zjs_port_ring_buf_put(&ring_buffer, (uint16_t)id, 0, (uint32_t*)args, (uint8_t)size / 4);
-        if (ret != 0) {
-            DBG_PRINT("error putting into ring buffer, ret=%u\n", ret);
-        }
-    } else {
-        // signal without args
-        uint32_t dummy = 0;
-        int ret = zjs_port_ring_buf_put(&ring_buffer, (uint16_t)id, 0, (uint32_t*)&dummy, (uint8_t)1);
-        if (ret != 0) {
-            DBG_PRINT("error putting into ring buffer, ret=%u\n", ret);
-        }
+    int ret = zjs_port_ring_buf_put(&ring_buffer, (uint16_t)id, 0, (uint32_t*)args, (uint8_t)((size + 3) / 4));
+    if (ret != 0) {
+        DBG_PRINT("error putting into ring buffer, ret=%u\n", ret);
```

Comment:
I totally agree, I meant to use ERR_PRINT, but I am still getting used to ERR_PRINT being a macro now days :)